clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java

```diff
@@ -1193,17 +1197,17 @@ public void waitForNodesHavingSystemStateVersionEqualToOrAbove(int version, int
                     context.log(logger, Level.INFO, ackedNodes + " nodes now have acked system state " + version + " or higher.");
                     return;
                 }
-                long remainingTime = maxTime - System.currentTimeMillis();
-                if (remainingTime <= 0) {
-                    throw new IllegalStateException("Did not get " + nodeCount + " nodes to system state " + version + " within timeout of " + timeout + " milliseconds.");
+                Duration remainingTime = Duration.between(Instant.now(), endTime);
+                if (remainingTime.isNegative()) {
```

Comment:
Prefer Instant.now().isAfter(endTime) as above for consistency and readability.

Reply:
👍